### PR TITLE
Add magnetometer

### PR DESCRIPTION
**Pull request by brooksben11**
## Proposed change

Add magnetometer as an option for capturing gas consumption

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

**Comment by brooksben11** [comment]
Any reason this hasn't been merged yet?

File: source/_docs/energy/gas.markdown [hunk 1]
```diff
@@ -29,3 +29,8 @@ We have worked with creator [Marcel Zuidwijk](https://www.zuidwijk.com) to devel
 
 ![Photo of the AI-on-the-edge-device Workflow](/images/docs/energy/ai-on-the-edge-device.jpg)
 
+#### Read the Gas Meter using a magnetometer
+
+[Diaphragm gas meters](https://en.wikipedia.org/wiki/Gas_meter#Diaphragm/bellows_meters) are the most common type of gas meter and their movement can frequently be observed with a magnetometer. The [QMC5883L](https://esphome.io/components/sensor/qmc5883l.html) is a common and inexpensive option that is supported by ESPHome. There are many posts on the forums of users having luck with this method, such as [this one](https://community.home-assistant.io/t/water-gas-meter-monitoring-via-magnetometer-sine-wave-to-pulse-issue/245904).
+
+![Photo of a typical diaphragm gas meter](https://en.wikipedia.org/wiki/File:Gas_meter.JPG)
```
Comment (frenck) [inline source/_docs/energy/gas.markdown]:
```suggestion
#### Read the Gas Meter using a magnetometer

[Diaphragm gas meters](https://en.wikipedia.org/wiki/Gas_meter#Diaphragm/bellows_meters) are the most common type of gas meter, and their movement can frequently be observed with a magnetometer. The [QMC5883L](https://esphome.io/components/sensor/qmc5883l.html) is a common and inexpensive option that ESPHome supports. Many posts on the forums of users having luck with this method, such as [this one](https://community.home-assistant.io/t/water-gas-meter-monitoring-via-magnetometer-sine-wave-to-pulse-issue/245904).

![Photo of a typical diaphragm gas meter](https://en.wikipedia.org/wiki/File:Gas_meter.JPG)
```

Comment (frenck) [inline source/_docs/energy/gas.markdown]: Images must be hosted from our own website.

**Review by frenck — CHANGES_REQUESTED** [review]

File: source/_docs/energy/gas.markdown (hunk 1 above)
Comment (brooksben11) [reply source/_docs/energy/gas.markdown]: Just spent 45 minutes trying to figure out how to add a local picture to this PR with no luck; just went ahead and deleted it altogether.

**Review by frenck — APPROVED** [review]
Thanks, @brooksben11 👍 

../Frenck

**Pull request merged**